CinemaCat.Api/Controllers/ImageController.cs

```diff
@@ -1,63 +1,36 @@
 {
-    private readonly BlobServiceClient _blobServiceClient;
-
-    public ImageController(BlobServiceClient blobServiceClient)
-    {
-        _blobServiceClient = blobServiceClient;
-    }
 
     [HttpGet("{id}", Name = "GetImage")]
     [ProducesResponseType(typeof(Stream), 200)]
-    public async Task<IActionResult> Get(Guid id, bool preview = false)
+    public async Task<ActionResult<Stream>> Get(Guid id, bool preview = false)
     {
-        BlobContainerClient containerClient = _blobServiceClient.GetBlobContainerClient("images");
-        var blob = containerClient.GetBlobClient((preview ? "compressed/" : "full/") + id);
-        var stream = await blob.OpenReadAsync();
-        return File(stream, "image/jpeg");
+        var req = new GetImageRequest { Id = id, IsPreview = preview };
+
+        var result = await mediator.Send(req);
+        return result.ToResult(x => File(x, "image/jpeg"), e => NotFound(e));
     }
 
     [HttpPost]
-    [ProducesResponseType(typeof(UploadImageResponse), 200)]
-    public async Task<IActionResult> UploadImage(IFormFile file)
+    [ProducesResponseType(typeof(UploadedImagesInfo), 200)]
+    public async Task<ActionResult<UploadedImagesInfo>> UploadImage(IFormFile file)
     {
-        var result = new UploadImageResponse();
-        BlobContainerClient containerClient = _blobServiceClient.GetBlobContainerClient("images");
-        var stream = file.OpenReadStream();
-        using (var fullImageStream = new MemoryStream())
-        using (var previewImageStream = new MemoryStream())
-        using (var image = Image.Load(stream))
-        {
-            image.Save(fullImageStream, new JpegEncoder());
-            var ratio = image.Height / image.Width;
-            var clone = image.Clone(x => x.Resize(250 * ratio, 250));
-            clone.Save(previewImageStream, new JpegEncoder());
-
-            previewImageStream.Position = 0;
-            fullImageStream.Position = 0;
-            var guid = Guid.NewGuid();
-            var blob = containerClient.GetBlockBlobClient("full/" + guid);
-            var blobCompressed = containerClient.GetBlockBlobClient("compressed/" + guid);
-            await blob.UploadAsync(fullImageStream, new BlobHttpHeaders { ContentType = "image/jpeg" });
-            await blobCompressed.UploadAsync(previewImageStream, new BlobHttpHeaders { ContentType = "image/jpeg" });
-
-            result.FullImageUrl = $"{Request.Scheme}://{Request.Host}{Url.RouteUrl("GetImage", new { id = guid })}";
-            result.CompressedImageUrl = $"{Request.Scheme}://{Request.Host}{Url.RouteUrl("GetImage", new { id = guid, compressed = true })}";
-        }
-
-        return Ok(result);
+        var req = new UploadImageRequest() { File = file };
+        var result = await mediator.Send(req);
+        var guid = result.Result.Id;
+        result.Result.FullImageUrl = $"{Request.Scheme}://{Request.Host}{Url.RouteUrl("GetImage", new { id = guid })}";
```

Comment:
https://learn.microsoft.com/en-us/aspnet/core/mvc/controllers/routing?view=aspnetcore-8.0#generating-urls-by-action-name let's try this one and see if it results in a shorter code

Reply:
done